react-blocks/src/blocks/Columns/Columns.js

```diff
@@ -0,0 +1,193 @@
+                    onChange={this.props.onColumnDescriptionChange.bind(this,index)}
+                  />
+
+                  { 'no_image' != columns_block_style &&
+                    <MediaPlaceholder
+                      labels={{ title: __('Column %s: Image', 'p4ge').replace('%s', index+1)}}
+                      icon="format-image"
+                      onSelect={this.props.onSelectImage.bind(this,index)}
+                      onSelectURL={this.props.onSelectURL.bind(this,index)}
+                      onError={this.props.onUploadError}
+                      accept="image/*"
+                      allowedTypes={["image"]}
+                    />
+                  }
+
+                  <TextControl
+                    label={__('Column %s: Button/CtA Link', 'p4ge').replace('%s', index+1)}
+                    placeholder={__('Enter link for column %s', 'p4ge').replace('%s', index+1)}
+                    value={item.cta_link}
+                    onChange={this.props.onCTALinkChange.bind(this,index)}
+                  />
+                  <ToggleControl
+                    label={__('Open link in new tab', 'p4ge')}
+                    help={__('Open Column %s: Button/CtA Link in a new tab', 'p4ge').replace('%s', index+1)}
+                    value={item.link_new_tab}
+                    checked={item.link_new_tab}
+                    onChange={this.props.onLinkNewTabChange.bind(this,index)}
+                  />
+                  <TextControl
+                    label={__('Column %s: Button/CtA Text', 'p4ge').replace('%s', index+1)}
+                    placeholder={__('Enter text of button/link for column %s', 'p4ge').replace('%s', index+1)}
+                    value={item.cta_text}
+                    onChange={this.props.onCTAButtonTextChange.bind(this,index)}
+                  />
+                </div>
+              );
+            })}
+
+            { 4 > columns.length && (
+              <Tooltip text={__('Add Column', 'p4ge')}>
+                <button
+                  className={
+                    "wp-block-p4ge-blocks-columns__addIcon"
+                  }
+                  onClick={this.props.addColumn}
+                >
+                  {__('Add Column', 'p4ge')} <Dashicon icon={"plus"} size={12} />
+                </button>
+              </Tooltip>
+            )}
+
+            { 0 < columns.length && (
+              <Tooltip text={__('Remove Column', 'p4ge')}>
+                <button
+                  className={
+                    "wp-block-p4ge-blocks-columns__removeIcon"
+                  }
+                  onClick={this.props.removeColumn}
+                >
+                  {__('Remove Column', 'p4ge')} <Dashicon icon={"minus"} size={12} />
+                </button>
+              </Tooltip>
+            )}
+          </div>
+        </Fragment>
+      );
+    }
+
+    render() {
+
+      // Temporary fix to refresh serversiderender component on columns fields changes.
```

Comment:
To fix the columns fields not updating onchange in `ServerSideRender ` component, I added below temporary fix.

Reply:
Yes, The random number trick works. 👍

Reply:
Wouldn't [componentDidUpdate](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/develop/react-blocks/src/blocks/Timeline/Timeline.js#L21) also be used? Or am I missing something?

Reply:
this is happening because changes on a sub-attribute of the columns attribute will not trigger component render in this case (but it works when a new element is removed/added to the array).
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/13/files#diff-2c222971ee3d2fcc0a3c7ce63a33916dR105

check here for a workaround. you'd have to clone the current array, set the sub-attribute and reset the columns attribute
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/develop/react-blocks/src/blocks/Submenu/SubmenuBlock.js#L129

Reply:
All of the options still feel like workarounds and "unreacty", I think we are missing something very simple here. I think we could revisit this after, maybe dispatching an action to WP's store... 🤔